lib/segment/src/index/field_index/map_index.rs

```diff
@@ -120,12 +122,13 @@ impl<N: Hash + Eq + Clone + Display + FromStr> MapIndex<N> {
         if self.point_to_values.len() <= idx as usize {
             self.point_to_values.resize(idx as usize + 1, Vec::new())
         }
-        self.point_to_values[idx as usize] = values.into_iter().map(|v| v.into()).collect();
-        for value in &self.point_to_values[idx as usize] {
-            let entry = self.map.entry(value.clone()).or_default();
-            entry.insert(idx);
 
-            let db_record = Self::encode_db_record(value, idx);
+        self.point_to_values[idx as usize].reserve(values.len());
```

Comment:
Please double -check the usage of `reserve` - it is reserving _additinal_ mem. We had problem with this method already

Reply:
Prefer `try_set_capacity` is applicable

Reply:
or `try_set_capacity_exact`

Reply:
fixed using `Vec::with_capacity()`